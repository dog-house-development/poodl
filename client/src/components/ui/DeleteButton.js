import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import ConfirmModal from './modal/ConfirmModal';

const propTypes = {
    onConfirm: PropTypes.func.isRequired,
    confirmQuestion: PropTypes.string.isRequired
};

export class DeleteButton extends Component {
    constructor(props) {
        super(props);
        this.state = {
            confirmModalOpen: false
        };
    }

    handleClick = () => {
        this.setState({ confirmModalOpen: true });
    };

    handleCancel = () => {
        this.setState({ confirmModalOpen: false });
    };

    handleConfirm = () => {
        this.setState({ confirmModalOpen: false });
        this.props.onConfirm();
    };

    getConfirmModal() {
        if (this.state.confirmModalOpen) {
            return (
                <ConfirmModal
                    title="Confirm Delete"
                    message={this.props.confirmQuestion}
                    onConfirm={this.handleConfirm}
                    onCancel={this.handleCancel}
                />
            );
        }
    }

    render() {
        return (
            <Button onClick={this.handleClick} size="small" className="delete-button" icon="remove_circle_outline">
                {this.props.children}
                {this.getConfirmModal()}
            </Button>
        );
    }
}

DeleteButton.propTypes = propTypes;
export default DeleteButton;
