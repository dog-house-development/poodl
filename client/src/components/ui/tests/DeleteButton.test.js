import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import DeleteButton from '../DeleteButton';

configure({ adapter: new Adapter() });

describe('DeleteButton tests', () => {
    let wrapper, instance;
    const setInstanceAndWrapper = (_props = {}) => {
        const props = _.assign(
            {},
            {
                onConfirm: () => {},
                confirmQuestion: 'Are you sure you want to delete this?'
            },
            _props
        );
        wrapper = shallow(<DeleteButton {..._.assign({}, props)} />);
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('render', () => {
        it('should render delete button correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
