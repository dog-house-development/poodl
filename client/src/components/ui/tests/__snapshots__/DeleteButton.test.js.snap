// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton tests render should render delete button correctly 1`] = `
<Button
  className="delete-button"
  icon="remove_circle_outline"
  kind="primary"
  onClick={[Function]}
  size="small"
  type="button"
/>
`;
